### PR TITLE
[FFM-1452]: Fix bool, int and number comparisons + implement slices

### PR DESCRIPTION
**Pull request by davejohnston**
This change fixes the bool, int and number comparisons.  It replaces
a cast with the appropriate function from the strconv package.

The slice type is implemented with support for in, contains and equals
operators.

Customers can store target attributes as strings, float64, bool and slices.
When creating a rule the value is always a slice of strings.

The code was trying to cast a string to a int, float etc which does not work.
This caused rules evaluations to fail.

Added unit tests to verify the type operators now work as expected.

### CODE REVIEW

File: types/bool.go [hunk 1]
```diff
@@ -17,61 +20,70 @@ func NewBoolean(value interface{}) (*Boolean, error) {
 	return nil, fmt.Errorf("%v: cant cast to a booelan", ErrWrongTypeAssertion)
 }
 
+// numberOperator iterates over value, convets it to a float64 and calls fn on each element
```
Comment (conormurray95) [inline types/bool.go:23]: could you give this comment an update

**Review by conormurray95 — APPROVED** [review]
LGTM

File: types/slice.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+		data: value,
+	}
+}
+
+// In compares the attributes held by the slice with the input values.
+// it will first determine what type of slice this is, before casting values
+// to the appropriate type.
+func (s Slice) In(values []string) bool {
+	// Determine what kind of slice we have
+	switch s.data.(type) {
```
Comment (enver-bisevac) [inline types/slice.go:27]: data is interface{} and that can be a pointer, what do you think do we need to check for pointer on slice?

Comment (davejohnston) [reply types/slice.go:27]: Yes - I'll add a test case for this as well.

File: types/bool.go (hunk 1 above)
Comment (davejohnston) [reply types/bool.go:23]: yep

**Review by enver-bisevac — APPROVED** [review]
LGTM

**Pull request merged**